spec.md

```diff
@@ -2103,6 +2229,47 @@ Condition | gRPC Code | Description | Recovery Behavior
 | Call not implemented | 12 UNIMPLEMENTED | NodeGetInfo call is not implemented by the plugin or disabled in the Plugin's current mode of operation. | Caller MUST NOT retry. Caller MAY call `ControllerGetCapabilities` or `NodeGetCapabilities` to discover Plugin capabilities. |
 
 
+#### `NodeExpandVolume`
+
+A Node Plugin MUST implement this RPC call if it has `EXPAND_VOLUME` node capability.
+This RPC call allows CO to expand volume on a node.
+
+`NodeExpandVolume` ONLY supports expansion of already node-published or node-staged volumes on the given `volume_path`.
+
+If plugin has `STAGE_UNSTAGE_VOLUME` node capability - `NodeExpandVolume` MUST be called after successful `NodeStageVolume`.
+`NodeExpandVolume` MAY be called before or after `NodePublishVolume` if plugin has `STAGE_UNSTAGE_VOLUME` node capability otherwise `NodeExpandVolume` MUST be called after successful `NodePublishVolume`.
+
+```protobuf
+message NodeExpandVolumeRequest {
+  // The ID of the volume. This field is REQUIRED.
+  string volume_id = 1;
+
+  // The path on which volume is available. This field is REQUIRED.
+  string volume_path = 2;
+
+  // This allows CO to specify the capacity requirements of the volume
+  // after expansion. If omitted the volume will be expanded to maximum
+  // available size. This field is OPTIONAL.
+  CapacityRange capacity_range = 3;
```

Comment:
I was thinking of filesystem types that force you to specify a size for expansion. In those cases, this parameter may be needed.  It could be argued that, plugin can probably find out available space and issue appropriate resize command but I think it might be explicit whenever necessary.

Reply:
That sounds like an optimization. Plugins that only support node-level expansion (maybe they don't even have a controller service) need this, so we can leave it.

Reply:
I wonder if we should include a clarifying comment along the lines of:

If capacity_range is omitted then a plugin MAY inspect the file system of the volume to determine the maximum capacity to which the volume can be expanded. In such cases a plugin MAY expand the volume to its maximum capacity.

Reply:
I have incorporated this suggestion.